MBI rework (#4386)
centreon/centreon-documentation: versioned_docs/version-25.10/reporting/rebuilding-data.md

* See [**ETL command reference**](#etl-command-reference).

### Before starting with the ETL

* Before starting with MBI, make sure you have [prepared your data as described here](preparing-data.md). Then, make sure the **gorgoned** process is running properly, and restart it if necessary.

   ```shell
   systemctl status gorgoned
   ```

   ```shell
   systemctl restart gorgoned 
   ```

* Before starting any rebuild operation, try to estimate the time needed for the rebuild ([look at the logs](troubleshooting.md#where-can-i-find-the-logs) to see how long execution takes). Depending on the volume of data on the central server, the rebuild process may take a long time. If you have a lot of data, to prevent duplicates or interruptions during the reconstruction, temporarily comment out the following line in the **/etc/cron.d/centreon-bi-engine** cron file to disable the daily execution of the ETL:

   ```shell
   #30 4 * * * root /usr/share/centreon-bi/bin/centreonBIETL -d >> /var/log/centreon-bi/centreonBIETL.log 2>&1
   ```

   Then restart crond:

   ```shell
   systemctl restart crond
   ```

>    > Don't forget to uncomment the line in the cron file and to restart **crond** after the rebuild is fully completed.

Reviewer: Maybe a link to reporting/rebuilding-data.md#after-running-the-rebuild-scripts can be added

* To stop the data retention manager from running at the same as the rebuild and causing problems, comment the line in **/etc/cron.d/centreon-bi-purge**.

   ```shell

>    30 7 * * 1-5 root /usr/share/centreon-bi//etl/dataRetentionManager.pl >> /var/log/centreon-bi//dataRetentionManager.log 2>&1

Reviewer: ```suggestion
   #30 7 * * 1-5 root /usr/share/centreon-bi//etl/dataRetentionManager.pl >> /var/log/centreon-bi//dataRetentionManager.log 2>&1
```

   ```

   Then restart crond:

   ```shell
   systemctl restart crond
   ```

>    > Don't forget to uncomment the line in the cron file and to restart **crond** after the rebuild is fully completed.

Reviewer: Maybe a link to reporting/rebuilding-data.md#after-running-the-rebuild-scripts can be added

### Rebuilding only events/availability or only metrics

The procedure described above is the standard, recommended procedure. However, if you have large amounts of data and you know for sure that you only need to rebuild events/availability or only metrics, perform the following steps in the table below:

* Only events/availability: steps 1, 3 and 4.
* Only metrics: steps 2, 3, 5 and 6.

Once rebuild is complete, [perform any necessary post-rebuild operations](#after-running-the-rebuild-scripts).

| Steps | Description | Command | Execution Time |
|------|-------------|---------|----------------|
| **1. Import event and availability data (excluding performance data)** | Import event data but not performance data (`data_bin`) between specific dates. (Specifically, use this command if there is a problem with the contents of the `mod_bam_reporting`, `hoststateevents` or `servicestateevents` tables). | `nohup /usr/share/centreon-bi/etl/importData.pl -r -s $date_start$ -e $date_end$ --ignore-databin --no-purge > /var/log/centreon-bi/rebuild_importDataEvents.log &` | **Fast** (minutes) |

> | **2. Import metrics (`data_bin`)** | Import only `data_bin` data between specific dates. | `nohup /usr/share/centreon-bi/etl/importData.pl -r --no-purge --databin-only -s $date_start$ -e $date_end$ > /var/log/centreon-bi/rebuild_importDataBin.log &` | **Fast** (minutes), depending on the number of days imported |

Reviewer: ```suggestion
| **2. Import metrics (`data_bin`)** | Import only `data_bin` data between specific dates. | `nohup /usr/share/centreon-bi/etl/importData.pl -r --no-purge --databin-only -s $date_start$ -e $date_end$ > /var/log/centreon-bi/rebuild_importDataBin.log &` | A few minutes to several hours |
```
